### DISCUSSION TITLE
What do we Need to Plan Before Starting the Rewrite?

### DISCUSSION THREAD

**Discussion by lyxal** [Vyxal/Vyxal#194] [Rewrite] [answered]
Last time we tried to do something major, it went all horribly wrong due to a lack of consensus/community understanding of what was happening. That ain't happening this time.

What do we need to make sure we have in place before we start rewriting everything? I assume that things like code style/documentation style and delegation of commands will need to be decided upon, but I'm probably missing a lot of things. 

Hence, I present to you a forum where you tell me what needs to happen. Alternatively, answer with what you think should happen.

**Comment by pxeger** [comment]
- Whoever is writing the transpiler needs to do it gradually enough (as in, committing often enough, not just in one go) that I can continually mirror its development to implement bytecode compilation

**Reply by ysthakur** [reply]
And it shouldn't be just one person - at least two people should be involved imo.

Also, I'd like to make an interpreter at the same time, so seconded.

**Comment by zoomlogo** [comment]
- **Coding Style**: As this is a big project, coding style will matter.

**Reply by pxeger** [reply]
just stick to what Black+Flake8 recommend

**Reply by ysthakur** [reply]
Those are mostly formatting and some warnings, but we do need to agree upon naming conventions and stuff. That said, I think we can just stick to the "old" Vyxal's coding style (mostly set by lyxal) and we'll be okay.

**Comment by Underslash12** [comment]
This seems pretty obvious - but we should definitely make a list of everything that we actually want to change, since continuously tacking on new things during it could delay / cause demotivation.

**Comment by ysthakur** [comment]
I suggest we make a rough wireframe first that everyone can live with. We need to plan out which files contain what kind of stuff (high level things only, such as "parser.py should create a lexer function and another function to use that tokenized output and create an AST", no real implementation details).

**Comment by pxeger** [comment, marked as answer]
Since the rewrite is finished, I'm closing this.
